test/src/unit-ctx.cc

```diff
@@ -103,3 +105,31 @@ TEST_CASE("C++ API: Test context tags", "[cppapi][ctx-tags]") {
     }
   }
 }
+
+TEST_CASE("C++ API: Test REST version endpoint", "[cppapi][rest][version]") {
+  tiledb::test::VFSTestSetup vfs_test_setup;
+  tiledb::Config config;
+  std::string serialization_format = GENERATE("JSON", "CAPNP");
+  config["rest.server_serialization_format"] = serialization_format;
+  config["rest.server_address"] = "http://127.0.0.1:8181";
+
+  // Only run these tests if the rest client has been initialized
+  if (vfs_test_setup.is_rest()) {
+    // Update the config in each section so that we always have a fresh Context.
+    // Otherwise, the JSON test will initialize rest_tiledb_version_ for CAPNP.
+    DYNAMIC_SECTION(
+        "GET request to retrieve REST tiledb version - "
+        << serialization_format) {
+      vfs_test_setup.update_config(config.ptr().get());
+      auto ctx = vfs_test_setup.ctx();
+      REQUIRE(ctx.ptr()->rest_client().get_rest_version() == "2.28.0");
+    }
+    DYNAMIC_SECTION(
```

Comment:
what is the difference between those 2 sections and they are, well, different sections? I see the same steps in both, they just check something different as last statement? I am missing something most probably :/

Reply:
The first SECTION tests the request in isolation, the second SECTION is to test that given a new Context with a RemoteRestClient, we will make the request on the first call to `rest_client().rest_version()`. I guess they are quite similar, I just felt the initialization on access was worth a SECTION.